src/neo4j_genai/retrievers/text2cypher.py

```diff
@@ -132,15 +132,22 @@ def get_search_results(
         except ValidationError as e:
             raise SearchValidationError(e.errors()) from e
 
-        if not self.custom_prompt:
-            prompt_template = Text2CypherTemplate()
-            prompt = prompt_template.format(
-                schema=self.neo4j_schema,
-                examples="\n".join(self.examples) if self.examples else "",
-                query=validated_data.query_text,
-            )
-        else:
-            prompt = self.custom_prompt
+        # parse the schema and examples inputs
+        examples_to_use = kwargs.get("examples") or (
```

Comment:
I think there was a reason to have examples and schema in the init and not in the search, @alexthomas93 maybe knows more about it?

Reply:
I like having the `examples` option in the search for the custom prompt because it allows for dynamic example selection.

Reply:
Ok, I see the point, thanks.